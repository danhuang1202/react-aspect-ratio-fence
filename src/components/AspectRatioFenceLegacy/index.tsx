import React, { FunctionComponent } from 'react'
import cns from 'classnames'

// @ts-ignore
import styles from './style.css'

interface Props {
  /**
   *  The aspect ratio of an image describes the proportional
   *  relationship between width and height.
   *   @default 1
   **/
  ratio: number
  /** Custom class name */
  customClass?: string
}

const AspectRatioFenceLegacy: FunctionComponent<Props> = ({
  ratio = 1,
  customClass,
  ...props
}) => {
  return (
    <div className={cns(styles.wrap, customClass)}>
      <div
        style={{
          paddingBottom: `${100 / ratio}%`
        }}
        className={styles.container}
        {...props}
      />
    </div>
  )
}

export default AspectRatioFenceLegacy
