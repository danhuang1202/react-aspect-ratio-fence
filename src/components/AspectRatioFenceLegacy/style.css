.wrap {
  position: relative;
  width: 100%;
}

.container {
  width: 100%;
  height: 0;
}

.container > :first-child {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
