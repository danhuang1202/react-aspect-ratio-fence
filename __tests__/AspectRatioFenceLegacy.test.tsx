import React from 'react'
import renderer from 'react-test-renderer'
import AspectRatioFenceLegacy from '@components/AspectRatioFenceLegacy'

describe('[snapshot] AspectRatioFence', () => {
  it('AspectRatioFenceLegacy', () => {
    const props = {
      ratio: 16 / 9
    }

    const dom = renderer.create(
      <AspectRatioFenceLegacy {...props}>
        <img alt="AspectRatioFence" src="AspectRatioFence.png" />
      </AspectRatioFenceLegacy>
    )
    expect(dom).toMatchSnapshot()
  })
})
