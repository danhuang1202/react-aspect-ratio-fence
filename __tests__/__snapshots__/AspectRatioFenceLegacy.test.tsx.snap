// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot] AspectRatioFence AspectRatioFenceLegacy 1`] = `
<div
  className="AspectRatioFenceLegacy_wrap_ZYFye"
>
  <div
    className="AspectRatioFenceLegacy_container_3VjLm"
    style={
      Object {
        "paddingBottom": "56.25%",
      }
    }
  >
    <img
      alt="AspectRatioFence"
      src="AspectRatioFence.png"
    />
  </div>
</div>
`;
